server/remote/gitea/gitea.go

```diff
@@ -381,7 +381,12 @@ func (c *Gitea) Activate(ctx context.Context, u *model.User, r *model.Repo, link
 	if err != nil {
 		return err
 	}
-	_, _, err = client.CreateRepoHook(r.Owner, r.Name, hook)
+	_, response, err := client.CreateRepoHook(r.Owner, r.Name, hook)
+	if (err != nil && response.StatusCode == 200) || response.StatusCode == 404 {
+		// if repo was renamed, Gitea redirects as a GET request which
+		// results in an error of the SDK which can not parse the response
+		return fmt.Errorf("Could not find repository")
```

Comment:
If `err != nil`, should the returned error `fmt.Errorf` include the `err`?

Reply:
I'll look into it

Reply:
I would do:
```go
if err != nil {
  if response != nil && response.StatusCode == 404 {
    return fmt.Errorf("Could not find repository")
  }
  return err
}

Reply:
redirects should have a different status code ...
you can chech tat also ...
```go
if err != nil {
  if response != nil {
    if response.StatusCode == 404 {
    return fmt.Errorf("Could not find repository")
    }
    if response.StatusCode == xyz {
      ... // or use switch case ...
    }
  }
  return err
}

Reply:
Please take a look at https://github.com/woodpecker-ci/woodpecker/pull/780/commits/5393195038a95d1913b71b23fbe5fe7617d4829f